app/src/main/kotlin/com/hippo/ehviewer/ui/main/AvatarIcon.kt

```diff
@@ -0,0 +1,178 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import arrow.core.Either
+import arrow.core.Some
+import arrow.core.left
+import arrow.core.none
+import arrow.core.right
+import coil3.compose.AsyncImage
+import com.hippo.ehviewer.R
+import com.hippo.ehviewer.Settings
+import com.hippo.ehviewer.client.EhEngine
+import com.hippo.ehviewer.client.EhUtils
+import com.hippo.ehviewer.client.parser.HomeParser
+import com.hippo.ehviewer.collectAsState
+import com.hippo.ehviewer.ui.tools.DialogState
+import com.hippo.ehviewer.util.displayString
+import com.ramcosta.composedestinations.navigation.DestinationsNavigator
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.conflate
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.merge
+import kotlinx.coroutines.flow.retryWhen
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import moe.tarsin.coroutines.runSuspendCatching
+
+private val limitScope = CoroutineScope(Dispatchers.IO)
+private val refreshEvent = MutableSharedFlow<Unit>()
+private suspend fun refresh() = refreshEvent.emit(Unit)
+private fun <T> Flow<T>.keepIf(predicate: (old: T, new: T) -> Boolean) = distinctUntilChanged(predicate)
+
+private val limitFlow = refreshEvent.conflate().map { EhEngine.getImageLimits().right() }
+    .retryWhen<Either<String, HomeParser.Result>> { e, _ -> emit(e.displayString().left()).let { true } }
+    .map(::Some).let { merge(it, refreshEvent.map { none() }) }
+    .keepIf { o, n -> o.isSome { it.isRight() } && n.isNone() }
+    .stateIn(limitScope, SharingStarted.Eagerly, none())
+
+context(CoroutineScope, DialogState, SnackbarHostState, DestinationsNavigator)
+@Composable
+fun AvatarIcon() {
+    val hasSignedIn by Settings.hasSignedIn.collectAsState()
+    if (hasSignedIn) {
+        val placeholder = stringResource(id = R.string.please_wait)
+        val resetImageLimitSucceed = stringResource(id = R.string.reset_limits_succeed)
+        val result by limitFlow.collectAsState()
+        IconButton(
+            onClick = {
+                launch {
+                    refresh()
+                    awaitConfirmationOrCancel(
+                        confirmText = R.string.reset,
+                        title = R.string.image_limits,
+                        confirmButtonEnabled = result.isSome { it.isRight { it.limits.resetCost != 0 } },
```

Comment:
If result changes while the dialog is shown, this won't change.
We should move the reset button into the dialog.

Reply:
Done